### DISCUSSION TITLE
How to Build an Object Storage System Using Seaweek and Lustre？

### DISCUSSION THREAD

**Discussion by doublewood123** [seaweedfs/seaweedfs#6036] [Q&A]
We currently have a Lustre file system running, and the client has a new requirement to access Lustre through the S3 protocol. Can we build an object storage system by sharing the Lustre directory with Seaweek?

**Comment by kmlebedev** [comment]
I think it is possible, but you need to compare it with other solutions and understand what the limitations and feasibility are?
For example
1. rclone s3 serve https://rclone.org/commands/rclone_serve_s3/ 
2. https://github.com/juicedata/juicefs
![image](https://github.com/user-attachments/assets/d874f83a-aa4e-41b9-93aa-cffdc8c9868d)

3. You can do the same and add Lustre as a data storadge backend.
https://github.com/seaweedfs/seaweedfs/tree/master/weed/storage/backend/rclone_backend

4. But maybe you want something else, for example directly connect metadata from Lustre to the weed filler + S3?

**Reply by doublewood123** [reply]
thanks，customer requests that files written through the object gateway can also be accessed by the Lustre client. I don't think the seaweedfs is possible.
